diff --git a/src/gtkutil.c b/src/gtkutil.c
index df537c515a..c8a96ba477 100644
--- a/src/gtkutil.c
+++ b/src/gtkutil.c
@@ -1303,11 +1303,8 @@ xg_create_frame_widgets (struct frame *f)
      X and GTK+ drawing to a pure GTK+ build.  */
   gtk_widget_set_double_buffered (wfixed, FALSE);
 
-#if ! GTK_CHECK_VERSION (3, 22, 0)
-  gtk_window_set_wmclass (GTK_WINDOW (wtop),
-                          SSDATA (Vx_resource_name),
-                          SSDATA (Vx_resource_class));
-#endif
+  gtk_window_set_role (GTK_WINDOW (wtop),
+                       SSDATA (Vx_resource_name));
 
   /* Add callback to do nothing on WM_DELETE_WINDOW.  The default in
      GTK is to destroy the widget.  We want Emacs to do that instead.  */
